### PR TITLE
Fixed deprecation, updated vulnerable library, regular cleanup

### PR DESCRIPTION
**Pull request by voroscsoki**

### CODE REVIEW

File: bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/updater/ThreadSafeCounter.kt
Comment (KosmX) [inline bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/updater/ThreadSafeCounter.kt:1]: Either keep these methods public (why not public, it doesn't hurt) or just inline these.  
I think I made these to have getAndSet methods

**Review by KosmX — CHANGES_REQUESTED** [review]
Otherwise, looks good

File: bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/updater/ThreadSafeCounter.kt
Comment (voroscsoki) [reply bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/updater/ThreadSafeCounter.kt:1]: I think inlining is fine as there is no additional benefit to a separate function.

Comment (voroscsoki) [reply bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/updater/ThreadSafeCounter.kt:1]: However, if you think this should be public, then fine. It wasn't used anywhere but I suppose there could be a good reason to have these be public.

**Review by KosmX — APPROVED** [review]
It looks good now :3

**Pull request merged**